### PR TITLE
Cancel rpc if ClientContext is re-used.

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

**Comment by yang-g** [comment]
Fix #486

File: src/cpp/client/client_context.cc [hunk 1]
```diff
@@ -71,13 +63,26 @@ void ClientContext::AddMetadata(const grpc::string& meta_key,
 
 void ClientContext::set_call(grpc_call* call,
                              const std::shared_ptr<ChannelInterface>& channel) {
+  {
```
Comment (ctiller) [inline src/cpp/client/client_context.cc]: Do we want to do this or just crash?

(I'd be ok with crashing)

Comment (yang-g) [inline src/cpp/client/client_context.cc]: The existing gpr assert will cause a crash, and I thought the bug is asking
for something nicer.

Comment (ctiller) [inline src/cpp/client/client_context.cc]: I thought the bug was asking for the crash (and if it was already, we could
just close it)... +Abhishek Kumar <abhikumar@google.com> any opinion?

**Comment by ctiller** [comment]
@a11r - let's finish this discussion tomorrow

**Comment by wuyuanyi135** [comment]
I think this should be elaborated in the documentation and the error thrown should be more meaningful. I was stuck in the weird assert exception and finally realized that I reused the client context by accident.

**Pull request closed without merging**